test/new-e2e/tests/agent-metric-logs/log-agent/listener/tcp_test.go

```diff
@@ -0,0 +1,112 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package listener
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metric-logs/log-agent/utils"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awsdocker "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/docker"
+
+	appslogger "github.com/DataDog/test-infra-definitions/components/datadog/apps/logger"
+	"github.com/DataDog/test-infra-definitions/components/datadog/dockeragentparams"
+
+	"github.com/docker/docker/api/types"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed testdata/tcp-compose.yaml
+var tcpCompose string
+
+type dockerSuite struct {
+	e2e.BaseSuite[environments.DockerHost]
+}
+
+func TestTCPListener(t *testing.T) {
+	t.Skip("Skipping as don't currently support using labels to spin up TCP/UDP listener")
+	e2e.Run(t, &dockerSuite{}, e2e.WithProvisioner(
+		awsdocker.Provisioner(
+			awsdocker.WithAgentOptions(
+				dockeragentparams.WithExtraComposeManifest("logger", appslogger.DockerComposeManifest.Content),
+				dockeragentparams.WithExtraComposeManifest("logger-tcp", pulumi.String(tcpCompose)),
+			))))
+}
+
+func (d *dockerSuite) TestLogsReceived() {
+	d.EventuallyWithT(func(c *assert.CollectT) {
+		agentReady := d.Env().Agent.Client.IsReady()
+		assert.True(c, agentReady)
+	}, 1*time.Minute, 5*time.Second, "Agent was not ready")
+	agentVersion := d.Env().Agent.Client.Version()
+	d.T().Logf("Testing Agent Version '%v'\n", agentVersion)
+	statusOutput := d.Env().Agent.Client.Status().Content
+	d.T().Logf("Agent status:\n %v", statusOutput)
+	containerID, err := d.getLoggerContainerID()
+	require.NoError(d.T(), err)
+	assert.NotEmpty(d.T(), containerID)
+
+	dc := d.Env().Docker.GetClient()
+
+	// Command to execute inside the container
+	cmd := []string{
+		"curl", "-v",
+		"--header", "\"Content-Type: application/json\"",
+		"--request", "POST",
+		"--data", "'{\"message\":\"bob\"}'",
+		"localhost:3333/",
+	}
+
+	// Prepare the execution configuration
+	execConfig := types.ExecConfig{
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          cmd,
+	}
+
+	// Create the execution request
+	execResponse, err := dc.ContainerExecCreate(context.Background(), containerID, execConfig)
+	require.NoError(d.T(), err)
+
+	// Run the command in the container
+	execID := execResponse.ID
+	execStartCheck := types.ExecStartCheck{}
+	respID, err := dc.ContainerExecAttach(context.Background(), execID, execStartCheck)
+	require.NoError(d.T(), err)
+
+	// Wait for command execution to complete
+	_, err = dc.ContainerExecInspect(context.Background(), execID)
+	assert.NoError(d.T(), err)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wonder if this could be generalised enough to be a variant of  the wrapping `Docker.Execute`, instead of returning the lower docker client. What do you think? Could be addressed separately

Reply:
Otherwise we could have the higher level `Docker` anonymously embedding the lower `docker.Client` so that it exposes all SDK methods

Reply:
Yeah, I'm hoping to look at generalising this and adding it as function to [remotehost_docker.go](https://github.com/DataDog/datadog-agent/pull/25456/files#diff-d4fc43409783962bd4e04f055124afde9c8f8d8bdc2ef6d997d765b6293bc3c7) or as a helper function in the test framework. Figured this test once enabled and I write a few more I'll have a better idea of what Docker commands I'll want quick/easy access to.

Reply:
I still can't believe you have to write so much code to execute a command on a target container :/